org.lflang/src/org/lflang/generator/c/CGenerator.xtend

```diff
@@ -394,88 +406,61 @@ class CGenerator extends GeneratorBase {
-            if (mainDef !== null) {
-                targetConfig.compileAdditionalSources.add(fileConfig.getSrcGenPath + File.separator + "core/platform/lf_windows_support.c")
-            }
-        } else if (OS.indexOf("nux") >= 0) {
-            // Linux support
-            coreFiles.add("platform/lf_POSIX_threads_support.c")
-            coreFiles.add("platform/lf_C11_threads_support.c")
-            coreFiles.add("platform/lf_POSIX_threads_support.h")
-            coreFiles.add("platform/lf_C11_threads_support.h")
-            coreFiles.add("platform/lf_linux_support.c")
-            coreFiles.add("platform/lf_linux_support.h")
-            coreFiles.add("platform/lf_unix_clock_support.c")
-            // If there is no main reactor, then compilation will produce a .o file requiring further linking.
-            if (mainDef !== null) {
-                targetConfig.compileAdditionalSources.add(fileConfig.getSrcGenPath + File.separator + "core/platform/lf_linux_support.c")
-            }
-        } else {
-            errorReporter.reportError("Platform " + OS + " is not supported")
-        }
+        
+        addPlatformFiles(coreFiles);
         
         
         // If there are federates, copy the required files for that.
         // Also, create the RTI C file and the launcher script.
         if (isFederated) {
-            coreFiles.addAll("rti.c", "rti.h", "federate.c", "federate.h", "clock-sync.h", "clock-sync.c")
-            createFederateRTI()
-            createLauncher(coreFiles)
+            coreFiles.addAll(  
+                "federated" + File.separator + "net_common.h", 
+                "federated" + File.separator + "federate.c", 
+                "federated" + File.separator + "federate.h", 
+                "federated" + File.separator + "clock-sync.h", 
+                "federated" + File.separator + "clock-sync.c"
+            );
+            createFederatedLauncher(coreFiles);
         }
-        
-        copyFilesFromClassPath("/lib/core", fileConfig.getSrcGenPath + File.separator + "core", coreFiles)
-        
-        copyTargetHeaderFile()
 
         // Perform distinct code generation into distinct files for each federate.
         val baseFilename = topLevelName
         
         var commonCode = code;
         var commonStartTimers = startTimers;
         var compilationSucceeded = true
+        val oldFileConfig = fileConfig;
         for (federate : federates) {
+            fileConfig = oldFileConfig;
             startTimeStepIsPresentCount = 0
             startTimeStepTokens = 0
             
-            
             // If federated, append the federate name to the file name.
             // Only generate one output if there is no federation.
             if (isFederated) {
                 topLevelName = baseFilename + '_' + federate.name // FIXME: don't (temporarily) reassign a class variable for this
+                fileConfig = new FedFileConfig(fileConfig, federate.name);
```

Comment:
Ah, here it is used. I guess in this case also the copy constructor makes sense. However, I think the copy constructor should not be placed in `FileConfig`, but in `FedFileConfig` so that we can create federated configs from a file config.

Reply:
I think a lot of variables assigned in the copy constructor are private. I didn't want to convert them to protected unnecessarily. Do you think that would be a better solution?

Reply:
I see no objection in making those variables protected. Shall we just do that and implement @cmnrd's suggestion?

Reply:
I think a copy constructor in `FileConfig` is a useful feature to have. I don't agree necessarily that we should remove this copy constructor. Can leaving it as is lead to a dangerous situation?

Reply:
I don't see it being a problem, no. Invoke the copy constructor of `FileConfig` inside the copy constructor of `FedFileConfig`? That way you also don't have to modify the visibility.

Reply:
Okay. Done in ebc62defca50d3d407245c1e9b112a0a64d75531.